src/piton/extractors/omop.py

```diff
@@ -0,0 +1,205 @@
+import datetime
+from typing import Mapping, Optional, Sequence
+
+from piton import Event
+from piton.extractors.csv import CSVExtractor
+
+OMOP_BIRTH = 4216316
+
+
+class _DemographicsConverter(CSVExtractor):
+    """Convert the OMOP demographics table to events."""
+
+    def get_patient_id_field(self) -> str:
+        return "person_id"
+
+    def get_file_prefix(self) -> str:
+        return "person"
+
+    def get_events(self, row: Mapping[str, str]) -> Sequence[Event]:
+        if row["birth_datetime"]:
+            birth = datetime.datetime.fromisoformat(row["birth_datetime"])
+        else:
+            year = 1900
+            month = 1
+            day = 1
+
+            if row["year_of_birth"]:
+                year = int(row["year_of_birth"])
+            else:
+                raise RuntimeError(
+                    "Should always have at least a year of birth?"
+                )
+
+            if row["month_of_birth"]:
+                month = int(row["month_of_birth"])
+
+            if row["day_of_birth"]:
+                day = int(row["day_of_birth"])
+
+            birth = datetime.datetime(year=year, month=month, day=day)
+
+        return [
+            # 4216316 is the OMOP birth code
+            Event(start=birth, code=4216316, event_type=row["load_table_id"])
+        ] + [
+            Event(
+                start=birth,
+                code=int(row[target]),
+                event_type=row["load_table_id"],
+            )
+            for target in [
+                "gender_concept_id",
+                "ethnicity_concept_id",
+                "race_concept_id",
+            ]
+            if row[target] != "0"
+        ]
+
+
+def _try_numeric(val: str) -> float | memoryview | None:
+    if val == "":
+        return None
+    try:
+        return float(val)
+    except ValueError:
+        return memoryview(val.encode("utf8"))
+
+
+@dataclasses.dataclass
+class _ConceptTableConverter(CSVExtractor):
+    """A generic OMOP converter for handling tables that contain a single concept."""
+
+    prefix: str
+
+    file_suffix: str = ""
+    concept_id_field: Optional[str] = None
+    string_value_field: Optional[str] = None
+    numeric_value_field: Optional[str] = None
+
+    def get_patient_id_field(self) -> str:
+        return "person_id"
+
+    def get_file_prefix(self) -> str:
+        if self.file_suffix:
+            return self.prefix + "_" + self.file_suffix
+        else:
+            return self.prefix
+
+    def _get_date(
+        self, date_field: str, row: Mapping[str, str]
+    ) -> Optional[datetime.datetime]:
+        """Extract the highest resolution date from the raw data."""
+        for attempt in (date_field + "time", date_field):
+            if attempt in row and row[attempt] != "":
+                return datetime.datetime.fromisoformat(row[attempt])
+
+        return None
+
+    def get_events(self, row: Mapping[str, str]) -> Sequence[Event]:
+        def normalize_to_float_if_possible(
```

Comment:
It is a bit weird that we have this function inside get_events function inside the class but we have the function _try_numeric that this function is dependent on entirely outside the class.

Reply:
_try_numeric is generic and doesn't depend on the exact logic of this class. normalize_to_float_if_possible does depend entirely on the internal logic here.